zebra-rpc/src/methods/get_block_template_rpcs.rs

```diff
@@ -278,6 +291,7 @@ where
         let mempool = self.mempool.clone();
         let latest_chain_tip = self.latest_chain_tip.clone();
         let mut state = self.state.clone();
+        let is_zebra_close_to_network_tip = self.sync_status.is_close_to_tip();
```

Comment:
This code runs when the future is created, which can be a long time from when the async body runs.

So if Zebra is under heavy load, the sync status here might be different from the sync status when the RPC call is executed.

So let's clone the sync_status instead, and check it near the start of the async block?

Reply:
Updated in [`471330d`](https://github.com/ZcashFoundation/zebra/pull/5769/commits/471330dc4769eaa7e00969c4b44b7d36382efb89).